analysis/src/nanoBase.cc

```diff
@@ -2,6 +2,18 @@
 
 using std::string;
 
+nanoBase::nanoBase(TTree *tree, Bool_t isMC) : Events(tree), m_isMC(isMC) {
```

Comment:
I worry here that we won't update this properly if eg m_lumi changes. Can we use something like a delegating constructor: https://stackoverflow.com/questions/308276/can-i-call-a-constructor-from-another-constructor-do-constructor-chaining-in-c instead (CMSSW should be able to use C++11 features), so it just passes to a single constructor that sets up the tools? Ditto for the massAnalyser code (though for that, remember, Dongjun will need the had info, so do something similar to vts code). But its great work, looking very nice!